controller/model/revocation_manager.go

```diff
@@ -0,0 +1,103 @@
+/*
+	Copyright NetFoundry Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package model
+
+import (
+	"github.com/openziti/edge/controller/persistence"
+	"github.com/openziti/edge/pb/edge_cmd_pb"
+	"github.com/openziti/fabric/controller/change"
+	"github.com/openziti/fabric/controller/command"
+	"github.com/openziti/fabric/controller/models"
+	"github.com/openziti/fabric/controller/network"
+	"github.com/openziti/storage/boltz"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+)
+
+func NewRevocationManager(env Env) *RevocationManager {
+	manager := &RevocationManager{
+		baseEntityManager: newBaseEntityManager[*Revocation, *persistence.Revocation](env, env.GetStores().Revocation),
+	}
+	manager.impl = manager
+
+	network.RegisterManagerDecoder[*Revocation](env.GetHostController().GetNetwork().Managers, manager)
+
+	return manager
+}
+
+type RevocationManager struct {
+	baseEntityManager[*Revocation, *persistence.Revocation]
+}
+
```

Comment:
You can avoid this if you use `RegisterCreateDecoder` and `RegisterDeleteDecoder` instead of `RegisterManagerDecoder`.